### DISCUSSION TITLE
Allow ring-color overwrite for focus/focus-visible

### DISCUSSION THREAD

**Discussion by sklawren** [tailwindlabs/tailwindcss#3868] [Ideas]
As per [my comment on Issue #52](https://github.com/tailwindlabs/tailwindcss-forms/issues/52#issuecomment-809025198)

The ring-color is currently fixed to blue-600, however, this is not overridable, and the ring color is not just the border color, it's also the focus box-shadow (and focus-visible). This creates a challenge for changing the color, especially for dark mode, where a single blue-600 color is not sufficient.

Please consider adding the ability to override the color in both standard and dark mode (and possibly other themes) so that it's easier to change the color than having to declare every single form element manually.

**Comment by adamwathan** [comment]
Hey! My comment in the original thread explains how to do this, just write some custom CSS that overrides the default CSS for form elements.

**Reply by stevensacks** [reply]
You limited your response to just border-color, but as pointed out, the focus is a box-shadow and it's different for different elements and it's a lot of work to override every single one.

Unless I'm missing something, it seems like it's a lot of work to do this.

**Reply by adamwathan** [reply]
It's the same amount of work as it would be to set up some default form styles without Tailwind by just writing CSS yeah, this plugin doesn't aim to do anything more than that right now.

**Comment by sklawren** [comment]
I forked the forms plugin and tried to implement this myself. It started getting overly complicated when I had to deal with focus-visible and dark mode (class vs media). I'm going to get back to it soon to come up with an elegant solution.

https://github.com/sklawren/tailwindcss-forms/blob/master/src/index.js#L256-L261

It allows for overriding the following colors:
- borderColor (also used to color the down caret svg in select)
- focusColor
- placeholderColor
- backgroundColor (for the `#fff` box-shadow backgrounds)

Implemented in `tailwind.config.js` like this:
```javascript
{
  extend: {
    forms: (theme) => ({
        DEFAULT: {
            borderColor: theme('colors.gray.300'),
            focusColor: theme('colors.blue.600'),
            placeholderColor: theme('colors.gray.500'),
        },
        dark: {
            backgroundColor: theme('colors.blue-gray.900'),
            borderColor: theme('colors.blue-grey.500'),
            focusColor: theme('colors.blue.300'),
            placeholderColor: theme('colors.blue-gray.500'),
        },
    }),
  }
}
```

**Reply by mrkldshv** [reply]
@sklawren I came across the same issue with trying to override default styles. Have you got any more elegant solution at the moment?

**Comment by jszobody** [comment]
@adamwathan Is it possible for this package to create a mapping like `formColor: theme('colors.blue')` in such a way that I could redefine that mapping in on own tailwind.config.js file to change the whole color scheme for all form elements? 

At the moment I've literally defined `blue: colors.amber` so that form elements match the rest of my amber colored app, but eww.

**Reply by adamwathan** [reply]
See previous discussion here:

https://github.com/tailwindlabs/tailwindcss-forms/issues/33#issuecomment-807883605
